presto-main/src/test/java/com/facebook/presto/operator/scalar/TestCanonicalizePhoneNumberFunction.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.type.VarcharType;
+import com.google.i18n.phonenumbers.PhoneNumberUtil;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+
+public class TestCanonicalizePhoneNumberFunction
```

Comment:
Can we test what happens on null inputs, case sensitivity of the format code, and on a country code that doesn't exist (and perhaps add a few more country codes)?

Reply:
Great points - I am planning to do the following:

1) Re: case sensitivity: I think it makes sense to uppercase the format and country code automatically to avoid some user churn? Thoughts? And I will add tests to ensure that both the cases work.

2) Re: null checks - Only one variant of this function accepts nulls, others given Presto's null policy - will just return null without even invoking the function definition. For that function I have both a NULL and non-NULL input to ensure that the function works for both.

3) Other countries - I wanted to draw a line where I am testing the function vs the library. My hope with this test is to ensure that the parameters being passed are actually going to the library and we are surfacing the errors up. For invalid country code - the library should throw an error. So I specifically tried this example - `"CANONICALIZE_PHONE_NUMBER('+91412 9 150.000 x 500', 'US', 'E164')"` where `+91` is the country code of India and the country that I pass in is `US` - so ideally I would have liked to throw an exception saying that don't specify country code explicitly if present in number or they should match - but this is how the library behaves - hence I wanted to call this out in the documentation.

Reply:
In point (3) - the example the library's output is that it ignores the `countrycode` if the provided string already has the encoded countrycode.